File: src/066/main.py
from math import *
max_val = -1
for i in range(2, 1001):
	root = int(sqrt(i))
	if root*root == i:
		continue
	b = root
	c = 1
	
	v = []
	h = []
	k = []
	
	s = set()
	v.append(b)
	p = (b, c)
	
	while p not in s:
		s.add(p)
		
		temp_c = (i - b*b) / c
		temp_b = b
		d = -1
		while -temp_b <= root:  
			temp_b -= temp_c;
			d+=1
		temp_b += temp_c
		v.append(d)
		
		b = -temp_b
		c = temp_c
		
		p = (b, c)
	
	h.append(1)
	h.append(0)
	h.append(1)
	k.append(0)
	k.append(1)
	k.append(0)
	
	aIndex = 0
	while True:
		h.append(v[aIndex]*h[len(h)-1]+h[len(h)-2])
		k.append(v[aIndex]*k[len(k)-1]+k[len(k)-2])
		
		aIndex= 1 if aIndex+1>=len(v) else aIndex+1
		
		if h[-1]*h[-1] - i*k[-1]*k[-1] == 1:
			max_val = max(max_val, h[-1])
			print '{}^2 \n - {:>3}*{}^2\n={}\n'.format(h[-1], i, k[-1], h[-1]*h[-1] - i*k[-1]*k[-1]) 
			break

print "D=661"
print max_val, 

File: src/066/main.cpp
#include <iostream>
#include <utility>
#include <algorithm>
#include <set>
#include <vector>
#include <math.h>


//LIMITS OF LONG LONG CAUSE ERRORS, USE PYTHON CODE
using namespace std;
int main(){	
	for(int i = 2; i <= 1000; i++) {
		int root = (int) sqrt(i);
		if(root*root == i) continue;
		int b = root;
		int c = 1;
		
		//cout << i << "\n";
		vector<long long> v, h, k;
		set<pair<int, int>> s;
		pair<int, int> p = make_pair(b, c);
		v.push_back(b);
		while(s.find(p) == s.end()){
			s.insert(p);
			
			int temp_c = (i - b*b) / c; 
			int temp_b = b;
			int d = -1;
			while(-temp_b <= root) {
				temp_b -= temp_c;
				d++;
			}
			temp_b += temp_c;
			v.push_back(d);
			
			b = -temp_b;
			c = temp_c;
			
			p = make_pair(b, c);
		}
		
		h.push_back(1);
		h.push_back(0);
		h.push_back(1);
		k.push_back(0);
		k.push_back(1);
		k.push_back(0);
		
		//for(int q: v){
		//	cout << q << " ";
		//}
		//cout << "\n";
		int aIndex = 0;
		int count = 10;
		while(true){
			//cout << h.back() << "/" << k.back() << "\n";
			h.push_back(v[aIndex]*h[h.size()-1]+h[h.size()-2]);
			k.push_back(v[aIndex]*k[k.size()-1]+k[k.size()-2]);
			aIndex++;
			if(aIndex >= v.size()) aIndex = 1;
		
			if(h.back()*h.back() - i*k.back()*k.back() == 1) {
				cout << h.back() << "\t\t\t^2 - " << i << "*" << k.back() << "^2 = 1\n";
				break; 
			}
		}
	}
	return 0;
}
